core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -950,69 +939,52 @@ private QueryModel parseDmlUpdate(GenericLexer lexer) throws SqlException {
-        CharSequence tok = tok(lexer, "table name or alias");
-        CharSequence tableName = GenericLexer.immutableOf(tok);
-        ExpressionNode tableNameExpr = ExpressionNode.FACTORY.newInstance().of(ExpressionNode.LITERAL, tableName, 0, 0);
-        updateQueryModel.setTableName(tableNameExpr);
-        fromModel.setTableName(tableNameExpr);
-
-        tok = tok(lexer, "AS, SET or table alias expected");
-        if (isAsKeyword(tok)) {
-            tok = tok(lexer, "table alias expected");
-            if (isSetKeyword(tok)) {
-                throw SqlException.$(lexer.lastTokenPosition(), "table alias expected");
-            }
-        }
-
-        if (!isAsKeyword(tok) && !isSetKeyword(tok)) {
-            // This is table alias
-            CharSequence tableAlias = GenericLexer.immutableOf(tok);
-            ExpressionNode tableAliasExpr = ExpressionNode.FACTORY.newInstance().of(ExpressionNode.LITERAL, tableAlias, 0, 0);
-            updateQueryModel.setAlias(tableAliasExpr);
-            tok = tok(lexer, "SET expected");
-        }
-
-        if (!isSetKeyword(tok)) {
-            throw SqlException.$(lexer.lastTokenPosition(), "SET expected");
-        }
-
-        while (true) {
-            // Column
-            tok = tok(lexer, "column name");
-            CharSequence col = GenericLexer.immutableOf(GenericLexer.unquote(tok));
-            int colPosition = lexer.lastTokenPosition();
-
-            expectTok(lexer, "=");
-
-            // Value expression
-            ExpressionNode expr = expr(lexer, (QueryModel) null);
-            ExpressionNode setColumnExpression = expressionNodePool.next().of(ExpressionNode.LITERAL, col, 0, colPosition);
-            updateQueryModel.getUpdateExpressions().add(setColumnExpression);
-
-            QueryColumn valueColumn = queryColumnPool.next().of(col, expr);
-            fromModel.addBottomUpColumn(valueColumn);
+    private void parseFromClause(GenericLexer lexer, QueryModel model, QueryModel masterModel) throws SqlException {
+        CharSequence tok = expectTableNameOrSubQuery(lexer);
+        // expect "(" in case of sub-query
 
+        if (Chars.equals(tok, '(')) {
+            QueryModel proposedNested = parseAsSubQueryAndExpectClosingBrace(lexer, masterModel.getWithClauses());
             tok = optTok(lexer);
-            if (tok == null) {
-                break;
-            }
 
-            if (tok.length() != 1 || tok.charAt(0) != ',') {
+            // do not collapse aliased sub-queries or those that have timestamp()
+            // select * from (table) x
+            if (tok == null || (tableAliasStop.contains(tok) && !SqlKeywords.isTimestampKeyword(tok))) {
+
+                final QueryModel target = proposedNested.getNestedModel();
+                // when * is artificial, there is no union, there is no "where" clause inside sub-query,
+                // e.g. there was no "select * from" we should collapse sub-query to a regular table
+                if (
```

Comment:
What if there's group by or joined model(s) contain where clause, union model, etc. ?

Reply:
yeah, this is taken into account. Full condition is below your comment:

```
          proposedNested.isArtificialStar()
                                && proposedNested.getUnionModel() == null
                                && target.getWhereClause() == null
                                && target.getOrderBy().size() == 0
                                && target.getLatestBy().size() == 0
                                && target.getNestedModel() == null
                                && proposedNested.getLimitLo() == null
                                && proposedNested.getLimitHi() == null
```

Reply:
Conditional checks if : 
- proposedNested doesn't have set operations & limit clause 
- target doesn't have where clause, order by, latest by or nested model 
Could proposedNested itself contain e.g. group by, joins, latest by that could disappear  because only tablename, alias, timestamp and joins are copied ?

Reply:
There are no existing unit tests that would need additional checks. I will try to see if these are actual possible combinations, considering how models are created and nested.

@bziobrowski if you already have an idea of SQLs' that might not work correctly with this feature, please share

Reply:
you were right, there were side effects with sample & group by. Fixed here: https://github.com/questdb/questdb/pull/2031/commits/52ca146d5f5c5ba1bae85d1106945040337c1378